bin/spack

```diff
@@ -62,3 +62,7 @@ if 'ruamel' in sys.modules:
 # Once we've set up the system path, run the spack main method
 import spack.main  # noqa
 sys.exit(spack.main.main())
+
+# Once we've set up the system path, run the spack main method
+if __name__ == "__main__":
+    sys.exit(spack.main.main())
```

Comment:
This does not seem to be the same as the logic in @scheibelp's PR

Reply:
https://github.com/spack/spack/blob/897700465bff418a414a9dbd2c12381b0b1cabfa/bin/spack

Reply:
Yes, your PR has `sys.exit(spack.main.main())` twice, his only has it once.

Reply:
Now you've undone his change altogether...

Reply:
Fixed, the error I get now seems to be related to a wrong initialization 

` ==> Error: object.__init__() takes exactly one argument (the instance to initialize) `